// buffetjs/dist/components/Select/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _invariant = _interopRequireDefault(require("invariant"));

var _input = require("../../commonPropTypes/input");

var _Select = _interopRequireDefault(require("../../styled/Select"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Select(_ref) {
  var autoFocus = _ref.autoFocus,
      id = _ref.id,
      name = _ref.name,
      onChange = _ref.onChange,
      options = _ref.options,
      tabIndex = _ref.tabIndex,
      value = _ref.value,
      rest = _objectWithoutProperties(_ref, ["autoFocus", "id", "name", "onChange", "options", "tabIndex", "value"]);

  var lengthOfReactElementsInOptions = options.filter(function (option) {
    return _react["default"].isValidElement(option);
  }).length;

  var formatOptions = function formatOptions() {
    return options.map(function (option) {
      return _react["default"].createElement("option", {
        key: JSON.stringify(option.value) || option,
        value: option.value || option
      }, option.label || option);
    });
  };

  var renderOptions = function renderOptions() {
    var areOptionsValidReactElements = options.every(function (option) {
      return _react["default"].isValidElement(option);
    });
    return areOptionsValidReactElements ? options : formatOptions();
  };

  if (lengthOfReactElementsInOptions > 0) {
    (0, _invariant["default"])(lengthOfReactElementsInOptions === options.length, 'You have mixed up React Elements and non React Elements in your options array ');
  }

  return _react["default"].createElement(_Select["default"], _extends({
    autoFocus: autoFocus,
    id: id || name,
    name: name,
    onChange: onChange,
    tabIndex: tabIndex,
    value: value
  }, rest), renderOptions());
}

Select.defaultProps = _objectSpread({}, _input.commonDefaultProps, {
  options: []
});
Select.propTypes = _objectSpread({}, _input.commonPropTypes, {
  options: _propTypes["default"].arrayOf(_propTypes["default"].oneOfType([_propTypes["default"].shape({
    label: _propTypes["default"].string,
    value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number, _propTypes["default"].object])
  }), _propTypes["default"].string, _propTypes["default"].node])),
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number, _propTypes["default"].object]).isRequired
});
var _default = Select;
exports["default"] = _default;
